frontend/ui/elements/screensaver_menu.lua

```diff
@@ -235,6 +236,39 @@ return {
                             },
                         },
                     },
+                    {
+                        text = _("Duration format"),
+                        sub_item_table = {
+                            {
+                                text_func = function()
+                                    local util = require("util")
+                                    -- sample text to show 1h23m
+                                    local return_text = util.secondsToHClock(4980, true)
+                                    return T(_("Modern (%1)"), return_text)
+                                end,
+                                checked_func = function()
+                                    return G_reader_settings:readSetting("screensaver_duration_format") == "modern"
+                                end,
+                                callback = function()
+                                    G_reader_settings:saveSetting("screensaver_duration_format", "modern")
+                                end,
+                            },
+                            {
+                                text_func = function()
+                                    local util = require("util")
+                                    -- sample text to show 1:23
+                                    local return_text = util.secondsToClock(4980, true)
+                                    return T(_("Classic (%1)"), return_text)
+                                end,
+                                checked_func = function()
+                                    return G_reader_settings:readSetting("screensaver_duration_format") == "classic"
+                                end,
+                                callback = function()
+                                    G_reader_settings:saveSetting("screensaver_duration_format", "classic")
+                                end,
+                            },
+                        }
+                    }
```

Comment:
Vaguely wondering if it wouldn't be worth merging all of these into a single setting in Date/Time, like what was done for 24h clocks?

Reply:
I had the same thought when I originally put this in. It would be nice to have a single user setting to control this, and a new util function that returns the clock format based on that user preference so plug-ins wouldn't need to do this logic. To do it right, I would change the plug-in's that use util.secondsToClock or util.secondsToHClock to use this new function, and move the relevant user settings to a new one within Device -> "time and date". It would touch readerfooter, screensaver, bookstatuswidget, readerprogress and statistics. I would also add unit tests for the new util function.

Does this sound reasonable? If so, let me know if you'd prefer I close this PR while I do that work, since it's going to be a bit larger of a change than I originally planned.

Reply:
That sounds perfect, yeah.

I'd be inclined to prefer delaying the PR as a whole, but no strong feelings on that ;).

Reply:
Latest iteration has this ready for review! Please check it out!